dowhy/gcm/independence_test/__init__.py

```diff
@@ -1 +1,28 @@
 from .kernel import kernel_based, approx_kernel_based
+
+
+def independence_test(X, Y, conditioned_on=None, method="kernel"):
+    """Performs a (conditional) independence test.
+    Three methods for (conditional) independence test are supported at the moment:
+
+    * `kernel`: Kernel-based (conditional) independence test.
+
+      * K. Zhang, J. Peters, D. Janzing, B. Schölkopf. *Kernel-based Conditional Independence Test and Application in Causal Discovery*. UAI'11, Pages 804–813, 2011.
+      * A. Gretton, K. Fukumizu, C.-H. Teo, L. Song, B. Schölkopf, A. Smola. *A Kernel Statistical Test of Independence*. NIPS 21, 2007.
+
+    * `approx_kernel`: Approximate kernel-based (conditional) independence test.
+
+      * E. Strobl, K. Zhang, S. Visweswaran. *Approximate kernel-based conditional independence tests for fast non-parametric causal discovery*. Journal of Causal Inference, 2019.
+
+    :param X: Observations of X.
+    :param Y: Observations of Y.
+    :param conditioned_on: Observations of conditioning variable if we want to perform a conditional independence test. By default, independence test is carried out.
+    :param method: Method for conditional independence test. By default kernel-based conditional independence test is used.
```

Comment:
for convenience, it will be good to mention the allowed values for `method` here. Can say: supported values are "kernel" and "approx_kernel". I realize that it is written at the top but users often may just look at the specific parameter's doc.